src/core/src/op/matmul.cpp

```diff
@@ -117,8 +106,10 @@ void ngraph::op::v0::MatMul::validate_and_infer_types() {
                           get_input_element_type(1),
                           ").");
 
-    const auto &A_shape = get_input_partial_shape(0), B_shape = get_input_partial_shape(1);
-    std::vector<ov::PartialShape> input_shapes = {A_shape, B_shape};
-    std::vector<ov::PartialShape> output_shapes = shape_infer(this, input_shapes);
+    const auto &A_shape = get_input_partial_shape(0), &B_shape = get_input_partial_shape(1);
```

Comment:
Should it be split into two lines?

Reply:
Changed, as in current version is easy make mistake between value and reference.